source/tools/testConsistency/ClientApp/app.test.js

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. 
+// To run these tests, first ensure that the flask app for nodeJs is running at 127.0.0.1 port 5000
+// and the flask app for .NET WPF is running at 127.0.0.1 port 4000. 
+const request = require("supertest");
+const app = require("./app");
+const fs = require("fs");
+
+test("The GET request to the index page should return a response code 200", function(done) {
+     request(app)
+      .get("/")
+      .then(function(response) {
+        expect(response.statusCode).toBe(200);
+        done();
+      });
+});
+
+test("Sending no card data should still return a response code 200", function(done) {
+    const templateBuffer = fs.readFileSync("../../../../samples/v1.0/Scenarios/ActivityUpdate.json");
+    const template = templateBuffer.toString('utf8')
+    request(app)
+      .post("/")
+      .send(cardTemplate=template, cardData="")
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending no card template should still return a response code 200", function(done) {
+    const data = {
+                    "name": "Matt Hidinger"
+                 }
+    request(app)
+      .post("/")
+      .send(cardTemplate="", cardData=data)
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending an empty form should still return a response code 200", function(done) {
+    const data = {
+                    "name": "Matt Hidinger"
+                 }
+    request(app)
+      .post("/")
+      .send(cardTemplate="", cardData="")
+      .then(function(response){
+          expect(response.statusCode).toBe(200);
+          done();
+      });
+});
+
+test("Sending invalid JSON data should still return a response code 200", function(done) {
+    const data = '{"title" : "invalid card data" ';
```

Comment:
We should actually test for invalid parameters as far as the actual POST is concerned - eg both null strings (which i am guessing is invalid, or an otherwise empty POST request body altogether (unsure if you're actually able to simulate that here using the test library?)

Reply:
I tested for null strings. There is a test there.  Regarding long strings, I tested some cards that were extremely long (the imageStretch under samples) and they do work